dash/_pages.py

```diff
@@ -0,0 +1,271 @@
+import os
+from os import listdir
+from os.path import isfile, join
+import collections
+from urllib.parse import parse_qs
+from . import _validate
+from ._utils import AttributeDict
+
+
+CONFIG = AttributeDict()
+PAGE_REGISTRY = collections.OrderedDict()
+
+
+def _infer_image(module):
+    """
+    Return:
+    - A page specific image: `assets/<title>.<extension>` is used, e.g. `assets/weekly_analytics.png`
+    - A generic app image at `assets/app.<extension>`
+    - A logo at `assets/logo.<extension>`
+    """
+    assets_folder = CONFIG.assets_folder
+    valid_extensions = ["apng", "avif", "gif", "jpeg", "png", "webp"]
+    page_id = module.split(".")[-1]
+    files_in_assets = []
+
+    if os.path.exists(assets_folder):
+        files_in_assets = [
+            f for f in listdir(assets_folder) if isfile(join(assets_folder, f))
+        ]
+    app_file = None
+    logo_file = None
+    for fn in files_in_assets:
+        fn_without_extension, _, extension = fn.partition(".")
+        if extension.lower() in valid_extensions:
+            if (
+                fn_without_extension == page_id
+                or fn_without_extension == page_id.replace("_", "-")
+            ):
+                return fn
+
+            if fn_without_extension == "app":
+                app_file = fn
+
+            if fn_without_extension == "logo":
+                logo_file = fn
+
+    if app_file:
+        return app_file
+
+    return logo_file
+
+
+def _filename_to_name(filename):
+    return filename.split(".")[-1].replace("_", " ").capitalize()
+
+
+def _infer_path(filename, template):
+    pages_folder = CONFIG.pages_folder.split("/")[-1]
+    if template is None:
+        path = (
+            filename.replace("_", "-").replace(".", "/").lower().split(pages_folder)[-1]
+        )
+        path = "/" + path if not path.startswith("/") else path
+        return path
+    # replace the variables in the template with "none" to create a default path if no path is supplied
+    path_segments = template.split("/")
+    default_template_path = ["none" if s.startswith("<") else s for s in path_segments]
+    return "/".join(default_template_path)
```

Comment:
In the `if template is None` block you ensure the final path starts with `/`, but here it look like it might not - should it?